pytition/petition/views.py

```diff
@@ -98,7 +98,12 @@ def detail(request, petition_id, do_confirmation=False, confirmation_hash=None):
                 do_subscribe = False
         except:
             do_subscribe = False
-
+        
+        if not (firstname or lastname):
+            errormsg = "Vous devez mettre votre nom et prénom pour que la signature soit valide"
```

Comment:
Soit le texte doit contenir "nom ou prénom", soit la logique au dessus est fausse.
firstname et lastname sont des strings, et si les champs sont vides se sont des chaines vides ''
or : 
```python
>>> '' or 'toto'
'toto'
>>> '' or ''
''
>>> not 'toto'
False
>>> not ''
True
```
Du coup la logique en place ne vaut True que si les 2 sont vides.
Dans le cas où tu veux forcer l'usage du nom ET prénom, je propose : 
```python
if '' in [firstname, lastname]:
```

Reply:
Oui j'ai oublié d'inverser la logique en réécrivant la ligne

Reply:
Perdu ça fait encore 2 commit, tu fais comment pour modifier ton commit sans en créer un nouveau depuis l'interface de github ?

Reply:
Pour squasher depuis l'interface je ne sais pas mais en console ça donne : 
```
$ git rebase -i HEAD~1
```
Ensuite sur le deuxième commit tu remplaces le mot ```pick``` par ```squash``` puis tu sauvegardes et tu quittes.
Ça va rouvrir un éditeur, là tu peux modifier le message de commit puis sauvegarder puis quitter.
Ensuite ça va créer un nouveau commit, qu'il va falloir "force push" vers le serveur via : 
```
$ git push origin +patch-4:patch-4
```

Sinon dans l'interface de pull request j'ai un bouton "squash and merge" mais ya des chances que ça perde l'authorship du commit.

Reply:
Je passe que par l'interface pour mes commits pour l'instant comme je fais que des bidouilles basiques. Du coup t'embêtes pas, tentes depuis l'interface de le faire, on verra bien ce que ça donne tqt ^^

Reply:
Ok c'est parti :)